contracts/registries/LicenseRegistryV2.sol

```diff
@@ -0,0 +1,382 @@
+        emit RoyaltyPolicyRegistered(royaltyPolicy);
+    }
+
+    function registerCurrencyToken(address token) external onlyProtocolAdmin {
+        _getLicenseRegistryStorage().registeredCurrencyTokens[token] = true;
+        emit CurrencyTokenRegistered(token);
+    }
+
+    function setExpireTime(address ipId, uint256 expireTime) external onlyLicensingModule {
+        _setExpireTime(ipId, expireTime);
+    }
+
+    function setMintingLicenseSpec(
+        address ipId,
+        address licenseTemplate,
+        uint256 licenseConfigId,
+        Licensing.MintingLicenseSpec calldata mintingLicenseSpec
+    ) external onlyLicensingModule {
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        if (!$.registeredLicenseTemplates[licenseTemplate]) {
+            revert Errors.LicenseRegistry__UnregisteredLicenseTemplate(licenseTemplate);
+        }
+        $.mintingLicenseSpecs[_getHash(ipId, licenseTemplate, licenseConfigId)] = Licensing.MintingLicenseSpec({
+            isSet: true,
+            mintingFee: mintingLicenseSpec.mintingFee,
+            mintingFeeModule: mintingLicenseSpec.mintingFeeModule,
+            receiverCheckModule: mintingLicenseSpec.receiverCheckModule,
+            receiverCheckData: mintingLicenseSpec.receiverCheckData
+        });
+
+        emit MintingLicenseSpecSet(ipId, licenseTemplate, licenseConfigId, mintingLicenseSpec);
+    }
+
+    function setMintingLicenseSpecForAll(
+        address ipId,
+        Licensing.MintingLicenseSpec calldata mintingLicenseSpec
+    ) external onlyLicensingModule {
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        $.mintingLicenseSpecsForAll[ipId] = Licensing.MintingLicenseSpec({
+            isSet: true,
+            mintingFee: mintingLicenseSpec.mintingFee,
+            mintingFeeModule: mintingLicenseSpec.mintingFeeModule,
+            receiverCheckModule: mintingLicenseSpec.receiverCheckModule,
+            receiverCheckData: mintingLicenseSpec.receiverCheckData
+        });
+        emit MintingLicenseSpecSetForAll(ipId, mintingLicenseSpec);
+    }
+
+    function attachLicenseConfigToIp(
+        address ipId,
+        address licenseTemplate,
+        uint256 licenseConfigId
+    ) external onlyLicensingModule {
+        if (!_existsLicenseConfig(licenseTemplate, licenseConfigId)) {
+            revert Errors.LicensingModule__LicenseConfigNotFound(licenseTemplate, licenseConfigId);
+        }
+
+        if (_isDerivativeIp(ipId)) {
+            revert Errors.LicensingModule__DerivativesCannotAddLicenseConfig();
+        }
+
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        if ($.expireTimes[ipId] < block.timestamp) {
+            revert Errors.LicenseRegistry__IpExpired(ipId);
+        }
+        $.attachedLicenseConfigs[ipId].add(licenseConfigId);
+    }
+
+    // solhint-disable-next-line code-complexity
```

Comment:
This warning is a little concerning, since it signals that the method is hard to review.
Please extract functionality to other methods (for example, checking for expiration could be a view function, the logic inside the for loop could be another method)

Reply:
The warning was due to multiple `if` clauses in the function, but it isn't complicated since all are simple check and revert errors. 
Anyway, separate functions to remove this warning.